solr/core/src/java/org/apache/solr/handler/admin/RequestSyncShardOp.java

```diff
@@ -93,6 +93,7 @@ public void execute(CallInfo it) throws Exception {
       }
     } finally {
       // no recoveryStrat close for now
+      // NOTE: try-with-resources is fine here; ok for `core` to be closed before `syncStrategy`
       if (syncStrategy != null) {
         syncStrategy.close();
```

Comment:
shouldn't `syncStrategy` be a try w/ resources inside `core != null` around line 58?

it seems weird at least looking at the change on github that syncStrategy is even closed outside of where its used?